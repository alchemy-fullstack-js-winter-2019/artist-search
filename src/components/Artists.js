import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default function Artists({ artists }) {
  const artistList = artists.map(artist => {
    return (
      <Link to={`/artist/${artist.id}`} key={artist.id}>
        <li key={artist.id}>{artist.name}</li>
      </Link>
    );
  });
  return (
    <ul>{artistList}</ul>
  ); 
}

Artists.propTypes = {
  artists: PropTypes.array.isRequired
}; 
