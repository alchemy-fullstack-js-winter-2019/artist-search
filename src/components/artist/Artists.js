import React from 'react';
import PropTypes from 'prop-types';
import { withPaging } from '../paging/Paging';

function Artists({ artists }) {
  const { listOfArtists } = artists;
  return (
    
  );
}

Artists.propTypes = { 
  artists: PropTypes.array.isRequired
};

export default withPaging(Artists);

