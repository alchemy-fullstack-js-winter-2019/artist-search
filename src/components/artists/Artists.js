// import React, { PureComponent } from 'react';
// import PropTypes from 'prop-types';
// import Artist from './Artist';
// import { getArtists } from '../../services/artistsApi';

// export class Artists extends PureComponent {
//   static propTypes = {
//     results: PropTypes.array.isRequired
//   };
//   render() {
//     const artists = this.props.results.map(artist => {
//       return (
//         <Artist key={artist.id} artist={artist}/>
//       );
//     });
//     return (
//       <ul>
//         {artists}
//       </ul>
//     );
//   }
// }

// Artists.propTypes = {
//   artist: PropTypes.array
// };

// export default Artists;
