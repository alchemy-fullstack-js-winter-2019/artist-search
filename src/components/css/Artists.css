.artists {
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
  li {
    padding: 5px;
    list-style: none;
  }
  li:hover {
    background: rgba(255, 243, 76, 0.781)
  }
  ul {
    padding-left: 0;
  }
}
