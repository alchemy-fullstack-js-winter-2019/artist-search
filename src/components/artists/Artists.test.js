import React from 'react';
import { shallow } from 'enzyme';
import Artists from './Artists';

describe('Artists', () => {
  it('renders a list of artists', () => {
    const wrapper = shallow(<Artists 
      artistText='Nirvana'
      page={1}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
