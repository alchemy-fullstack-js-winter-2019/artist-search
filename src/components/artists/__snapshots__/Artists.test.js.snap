// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists matches a snapshot 1`] = `
<ul>
  <a
    href="/artist/5b11f4ce-a62d-471e-81fc-a69a8278c7da"
    onClick={[Function]}
  >
    <li>
      <h2>
        Nirvana
      </h2>
      <h3>
        US
      </h3>
    </li>
  </a>
  <a
    href="/artist/9282c8b4-ca0b-4c6b-b7e3-4f7762dfc4d6"
    onClick={[Function]}
  >
    <li>
      <h2>
        Nirvana
      </h2>
      <h3>
        GB
      </h3>
    </li>
  </a>
  <a
    href="/artist/c49d69dc-e008-47cf-b5ff-160fafb1fe1f"
    onClick={[Function]}
  >
    <li>
      <h2>
        Nirvana
      </h2>
      <h3>
        FR
      </h3>
    </li>
  </a>
</ul>
`;
